code/lib/core-server/src/utils/watch-story-specifiers.ts

```diff
@@ -15,34 +14,58 @@ const isDirectory = (directory: Path) => {
   }
 };
 
-// Watchpack (and path.relative) passes paths either with no leading './' - e.g. `src/Foo.stories.js`,
-// or with a leading `../` (etc), e.g. `../src/Foo.stories.js`.
-// We want to deal in importPaths relative to the working dir, so we normalize
-function toImportPath(relativePath: Path) {
-  return relativePath.startsWith('.') ? relativePath : `./${relativePath}`;
+// Takes an array of absolute paths to directories and synchronously returns
+// absolute paths to all existing files and directories nested within those
+// directories (including the passed parent directories).
```

Comment:
Does this need to be sync? If we do it async might it give other parts of our boot up a chance to do work while waiting on FS?

Reply:
I don't think it _must_ be sync, we'd just create the watcher asynchronously after directory walking completes. IMO performance implications here are not significant tho, even with 10k directories readdirSync calls seem to complete in negligible time.